### PR TITLE
ci: add mergify to manage PRs

### PR DESCRIPTION
**Pull request by Xunzhuo**
WIP

### CODE REVIEW

**Comment by Xunzhuo** [comment]
@Mergifyio rebase

**Comment by Xunzhuo** [comment]
@Mergifyio refresh

**Comment by Xunzhuo** [comment]
@Mergifyio update

**Comment by Xunzhuo** [comment]
@Mergifyio rebase

**Comment by Xunzhuo** [comment]
@Mergifyio squash ci: add mergify ci

File: .github/workflows/mergify.yml
Comment (kebe7jun) [inline .github/workflows/mergify.yml:12]: can not read from CODEOWNERS?

**Comment by kebe7jun** [comment]
I see the official example, it place mergify.yml in root dir, dose this also supported?

File: .github/workflows/mergify.yml
Comment (Xunzhuo) [reply .github/workflows/mergify.yml:12]: Yes, mergify can just read from team, besides this, I do not know any optional actions to do this.

**Comment by Xunzhuo** [comment]
> I see the official example, it place mergify.yml in root dir, dose this also supported?

Yes

File: .github/workflows/mergify.yml
Comment (kebe7jun) [reply .github/workflows/mergify.yml:12]: So if it's a pr from another team, will it work after they approve it?

**Review by kebe7jun — APPROVED** [review]
No more hesitation, let's give it a try!

**Comment by kebe7jun** [comment]
@Mergifyio merge

**Comment by kebe7jun** [comment]
it seems that we need merge this pr first.

**Pull request merged**
